_includes/manuals/2.0/5.1.12_dynflow_scaling.md

```diff
@@ -0,0 +1,172 @@
+May 14 11:24:21 modest-gator dynflow-sidekiq@worker-2[1423]: 2020-05-14T11:24:21.449Z 1423 TID-gpcyculz7 INFO: Booting Sidekiq 5.2.8 with redis options {:id=>"Sidekiq-server-PID-1423", :url=>"redis://localhost:6379/0"}
+```
+
+### Scaling down
+Scaling down is just reverse process, let's say having `worker`, `worker-1` and `worker-2` was
+an overkill and we think we would be fine with just `worker` and `worker-1`.
+```
+root@modest-gator:/etc/foreman/dynflow# systemctl disable --now dynflow-sidekiq@worker-2
+Removed /etc/systemd/system/multi-user.target.wants/dynflow-sidekiq@worker-2.service.
+
+root@modest-gator:/etc/foreman/dynflow# rm /etc/foreman/dynflow/worker-2.yml
+```
+
+### Dedicating a worker to a queue
+For this we need to create a new configuration for a worker (or just copy an existing one and change the relevant lines)
+
+Create the configuration file, here we create a worker called `rex` having 5 threads and consuming items from the `remote_execution` queue
+```
+root@modest-gator:/etc/foreman/dynflow# cat <<EOF >/etc/foreman/dynflow/rex.yml
+:concurrency: 5
+:queues:
+  - remote_execution
+EOF
+```
+
+Now let's start the service and check its status
+```
+root@modest-gator:/etc/foreman/dynflow# systemctl start dynflow-sidekiq@rex
+
+root@modest-gator:/etc/foreman/dynflow# systemctl status dynflow-sidekiq@rex
+● dynflow-sidekiq@rex.service - Foreman jobs daemon - rex on sidekiq
+   Loaded: loaded (/lib/systemd/system/dynflow-sidekiq@.service; disabled; vendor preset: enabled)
+   Active: active (running) since Thu 2020-05-14 11:32:01 UTC; 3s ago
+     Docs: https://theforeman.org
+ Main PID: 1785 (ruby2.5)
+    Tasks: 5 (limit: 38437)
+   Memory: 115.8M
+   CGroup: /system.slice/system-dynflow\x2dsidekiq.slice/dynflow-sidekiq@rex.service
+           └─1785 ruby2.5 /usr/share/foreman/vendor/ruby/2.5.0/bin/sidekiq -e production -r /usr/share/foreman/extras/dynflow-sidekiq.rb -C /etc/foreman/dynflow/rex.yml
+
+May 14 11:32:01 modest-gator systemd[1]: Started Foreman jobs daemon - rex on sidekiq.
+May 14 11:32:02 modest-gator dynflow-sidekiq@rex[1785]: 2020-05-14T11:32:02.789Z 1785 TID-gpaz8n321 INFO: GitLab reliable fetch activated!
+May 14 11:32:02 modest-gator dynflow-sidekiq@rex[1785]: 2020-05-14T11:32:02.790Z 1785 TID-gpazi1pg1 INFO: Booting Sidekiq 5.2.8 with redis options {:id=>"Sidekiq-server-PID-1785", :url=>"redis://localhost:6379/0"}
+```
+
+### Tuning number of threads per worker
+The only thing that needs to be done is changing the number at the `:concurrency:` line and restarting the service.
+Either open it up in your favourite `$EDITOR` or do it with `sed`.
+
+Here we can see the worker from previous example has 5 threads.
+```
+# 1785 is the process id taken from previous invocation of systemctl status
+root@modest-gator:/etc/foreman/dynflow# ps -fp 1785
+foreman     1785       1  0 11:32 ?        00:00:14 sidekiq 5.2.8  [0 of 5 busy]
+```
+
+Next we change it to 15
```

Comment:
This is bad documentation honestly, we explain the process but not why you would want to do this and what are expected results. I know this is probably something our perf team should confirm, then let's resivist this later on.l

Reply:
This applies to this entire chapter. I'm afraid I don't have enough data to back any claims around why anyone should do this.